src/components/application_manager/include/application_manager/request_controller.h

```diff
@@ -152,16 +154,17 @@ class RequestController {
   * @param mobile_correlation_id Active mobile request correlation ID
   *
   */
-  void OnMobileResponse(const uint32_t& mobile_correlation_id,
-                        const uint32_t& connection_key);
+  void OnMobileResponse(const uint32_t mobile_correlation_id,
+                        const uint32_t connection_key,
+                        const int32_t function_id);
 
   /**
   * @brief Removes request from queue
   *
   * @param mobile_correlation_id Active mobile request correlation ID
   *
   */
-  void OnHMIResponse(const uint32_t& correlation_id);
+  void OnHMIResponse(const uint32_t correlation_id, const int32_t function_id);
```

Comment:
@AByzhynar function ID can be -1 (INVALID_VALUE) Please try to post all suggestions in one review iteration.